### PR TITLE
Foreign pod deletion transitory state

### PR DESCRIPTION
**Pull request by mlavacca**
# Description

Whenever a foreign pod is deleted in the foreign cluster, a new pod is created by the foreign `replicaset`. This situation leads to a transitory state in which there are two foreign pods mapped to the same local one:
* old marked for deletion pod in `terminating` status
* new pod in `pending`, `containerCreating`, or `running` status

If we simply reflect the status of the foreign pod to the local one, there are fluctuations between the old pod and the new one. To face this issue, when a pod is marked for deletion, we blacklist that pod in the incoming reflector, such that the old pod will not be handled anymore by the reflector, except for the deletion routine, that will remove the object from the blacklist map, freeing the occupied space, and allowing the mechanism to be efficient and sustainable at scale.

Address #380

### CODE REVIEW

File: pkg/virtualKubelet/apiReflection/reflectors/incoming/pods.go [hunk 1]
```diff
@@ -149,3 +162,21 @@ func (r *PodsIncomingReflector) CleanupNamespace(namespace string) {
 		}
 	}
 }
+
+func (r *PodsIncomingReflector) isAllowed(ctx context.Context, obj interface{}) bool {
```
Comment (palexster) [inline pkg/virtualKubelet/apiReflection/reflectors/incoming/pods.go:166]: I think that a comment to this method may help to understand this part.

File: pkg/virtualKubelet/apiReflection/reflectors/blacklists.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package reflectors
+
+import (
+	apimgmt "github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection"
+)
+
+type blackList map[string]struct{}
+
```
Comment (palexster) [inline pkg/virtualKubelet/apiReflection/reflectors/blacklists.go:8]: I think a comment to this function would help to understand how the blacklist works.

File: pkg/virtualKubelet/apiReflection/reflectors/incoming/pods.go [hunk 3]
```diff
@@ -111,8 +121,11 @@ func (r *PodsIncomingReflector) preAddUpdate(foreignPod *corev1.Pod) *corev1.Pod
 	return homePod.(*corev1.Pod)
 }
 
-// TODO: add here custom unavailable status
-func (r *PodsIncomingReflector) PreDelete(_ interface{}) interface{} {
+func (r *PodsIncomingReflector) PreDelete(obj interface{}) interface{} {
```
Comment (palexster) [inline pkg/virtualKubelet/apiReflection/reflectors/incoming/pods.go:124]: A short comment to specify what is done here with the blacklist may be useful.

**Review by palexster — APPROVED** [review]
/lgtm

**Review by AbakusW — APPROVED** [review]

**Pull request merged**